### ISSUE TITLE
Improve search method for identifying installers active in a county

### ISSUE THREAD

**Issue by ghost** [dhanur/Sunible-Beta#18]
Currently the code uses the following method to determine which installers to show on Dashboard:

if (zipCode.charAt(0) == '9')
    { // CALIFORNIA zip
        if (zipCode.charAt(1) == '3' || zipCode.charAt(1) == '5')
        { // FRESNO or SANTA CLARA zip
            // show social proof with view dashboard button
            getSocialProofDataByZipCode(zipCode, function(data){
                if (!data || !data[0])
                {
                    showNoDataFoundPage();
                }
                else
                {
                    changePage($currentPage, pages.socialProof);
                    initSocialProof(data, 'fresno');
                }
            });
            // dashboard
            getInstallersByZipCode(zipCode, function(data){
                if (data != undefined && data.length > 0)
                {
                    initDashboard(zipCode, data);
                }
            });
        }

This is not accurate enough and won't scale as we expand to all the counties in California.

**Comment by ghost** [comment]
Proposal:

Create a new function zipTest.php that takes the zip code entered by the user on the landing page as an input parameter.  The function returns two string values:  state, county. 

 zipTest.php is called before the decision on which page to display after the landing page is made.

Case 1:  if state = California AND county = Fresno, then display social proof page with Show Solar Providers button 
Case 2:  if state = California AND != Fresno, then display social proof page with Sign Me Up button
Case 3:  If state != California, then display sign me up modal dialog box

The first two cases will require another call to function getSocialProofDataByZipCode.php to get zip code data points for map.

zipTest.php will query the complete US zip code map to allow for future expansion to other states.

Alternate Proposal:

Call function getSocialProofDataByZipCode.php before making the decision on which screen to display next after landing page.  If the return array 'County" value =" Fresno", then Case 1.  If the return array 'County" value !=" Fresno" AND != NULL, then Case 2.  If the return array 'County" value =NULL, then Case 3.

**Comment by ghost** [comment]
Added function zipToCounty() and pushed the following code to QA  and to sunible.com.

Test Form:  http://sunible.com/zipToCountyTest.html

Enter in any US zip code and State, County will be returned.  Works for all US zip codes.

Function called:  zipToCountyTest.php which intiates zipToCounty() function in sunible_fun.php

**Comment by rlytvynchuk** [comment]
Delegated to John, 'cause trunk doesn't has zipToCountyTest.php file in repository. Neither master or beta. Please merge it to trunk.

When done, to test is it working on FE or not, you can call from console

getCountyFromZip('93210', function(data){
    console.log(data);
});

**Comment by rlytvynchuk** [comment]
John, here is the response I get calling getCountyFromZip('93210')

Fatal error: Call to a member function fetch_object() on a non-object in /Applications/XAMPP/xamppfiles/htdocs/sunible.local/sunible_fn.php on line 16

Using http://localhost/zipToCountyTest.php returns the same

**Comment by ghost** [comment]
Roman,

It was a name mismatch.  I changed the function names to match yours.
 Changes pushed to Trunk/origin.

zipToCountyTest.html - test harness
getCountyFromZip.php - function called to get county a specific zip code is
in
getCountyFromZip - function name within sunible_fn.php

John Bissell
CTO

```
        Solar Simplified

Cell:     (510) 229-0577
Skype:  john.bissell
Email:   john@sunible.com

55 Harrison Street, 3rd Floor
Oakland, CA 94607
```

**Comment by rlytvynchuk** [comment]
request: /getCountyFromZip.php?zip=93210

Fatal error: Call to a member function fetch_object() on a non-object in /Applications/XAMPP/xamppfiles/htdocs/sunible.local/sunible_fn.php on line 16

same for zipToCountyTest.html

**Comment by swenson86** [comment]
It looks like its searching for a local file in John or Roman's testing
environment rather than a dynamic file in a directory. I can't say what it
is for sure but that's just my thought!

**Comment by ghost** [comment]
zipcode_usa.sql too big to attach here so I uploaded to Dropbox folder.  You will need to add this table to the torrohol_ca_csi database to allow the getCountyFromZip function to work.

**Comment by rlytvynchuk** [comment]
I imported zipcode_usa.sql and it works. Assigned item to myself.

Sveta, please do import to your database.

**Comment by dhanur** [comment]
Roman's suggestion: Store logic at DB level rather than on Front-end. In other words, Zip Code sent to server, server sends back action "1 for Fresno", "2 for Contra Costa" etc. It will be x number of actions for x number of unique pages that will be served up. Roman will have an array on the FE to match.

**Comment by ghost** [comment]
Correct, although my understanding is that this is the preferred design to be delivered later.  In order to get this feature implemented this week, Roman will code a workaround in the front-end for the 16 counties.  I just uploaded the new installer table with the additional records so Sveta and Roman can test.  Sveta will notify you when QA is done so you can push to production.

**Comment by rlytvynchuk** [comment]
Done with Issue #51 . Under done I mean the Front-End logic to detect whether State and County is covered or not and provide the proper scenario.

Since it's more refactoring than bug, Assigned to Dhanur
